### PR TITLE
Code Insights: Include backend insights in insights count metric/ping

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/24177
Closes https://github.com/sourcegraph/sourcegraph/issues/24178

### Context
Historically we had been pinging the only extension-based insights which are stored as top-level properties in the setting cascade subjects. But since we introduced Backend Insights which are stored under the `insights.allrepos` key in subjects settings we should take into account this info too to be able to count insights properly. 

This PR adds logic to count backend insights with extension-based insights and at the same time extend ping counting event with new explicit info about extension and backend-based insights. 

```ts
interface InsightGroups {
    codeStatsInsights: number
    
    // All search insights extension-based + backend-based
    searchBasedInsights: number
    
    // Explicit number of backend based insights
    searchBasedBackendInsights: number
    
    // Explicit number of extension based insights
    searchBasedExtensionInsights: number
}
```
 
Note that all insights are counted only from shared places like org-level settings or global site-wide settings. Personal insights don't count. This logic about this was introduced in [this RFC about Code Insights Pings](https://docs.google.com/document/d/1YFmBiqeAZJccXJi8aFPdtvrP-ysLh46sePKH8RGiT6U/edit#)

> of current insights that are org-visible by week, by type of insight – the # of insights that are org-wide in visibility (note: if there are 4 one week, then one more is created, the next week of data should show 5. This metric is not measuring an org-visible insight just on the week it is created) 
This helps us understand if folks are sharing insights at all or just experimenting on their own

### CODE REVIEW

File: client/web/src/insights/core/analytics.ts [hunk 1]
```diff
@@ -231,31 +233,15 @@ export function parseInsightSettingsKey(key: string): Omit<InsightData, 'configu
         return undefined
     }
 
-    switch (insightType) {
-        // Exceptional extensions that don't adhere to the standard insight settings key format
-        case 'codeStatsInsights': {
-            if (key.split('.')[1] !== 'query') {
```
Comment (vovakulikov) [inline client/web/src/insights/core/analytics.ts:237]: @Joelkw It turned out that this logic was related and counted only old API code stats insights. After this PR change, it should be right counted. Could fix this related issue about code stats insights https://github.com/sourcegraph/sourcegraph/issues/24178

File: client/web/src/insights/core/analytics.ts [hunk 2]
```diff
@@ -113,44 +110,49 @@ export function getDaysFromInsightStep(step: Duration): number {
...
+ */
+export function getInsightsGroupedByType(settingsCascade: SettingsCascade<Settings>): InsightGroups {
     const { subjects } = settingsCascade
-    const {
-        organizations: { nodes: orgs },
-    } = authUser
-    const orgIDs = new Set(orgs.map(org => org.id))
 
-    const orgSubjects = subjects.filter(subject => orgIDs.has(subject.subject.id))
+    const globalSubjects = subjects.filter(configuredSubject => configuredSubject.subject.__typename === 'Site')
```
Comment (valerybugakov) [inline client/web/src/insights/core/analytics.ts:124]: Is it possible to use subject type-guards here?

```suggestion
    const globalSubjects = subjects.filter(configuredSubject => isGlobalSubject(configuredSubject.subject))
```

File: client/web/src/insights/core/analytics.ts [hunk 3]
```diff
@@ -113,44 +110,49 @@ export function getDaysFromInsightStep(step: Duration): number {
...
+    const orgSubjects = subjects.filter(configuredSubject => configuredSubject.subject.__typename === 'Org')
```
Comment (valerybugakov) [inline client/web/src/insights/core/analytics.ts:125]:
```suggestion
    const orgSubjects = subjects.filter(configuredSubject => isOrganizationSubject(configuredSubject.subject))
```

File: client/web/src/insights/core/analytics.ts [hunk 4]
```diff
@@ -113,44 +110,49 @@ export function getDaysFromInsightStep(step: Duration): number {
...
+
+        return mergedSettings ?? finalSettings
+    }, {} as Settings)
+
+    const codeStatsInsightCount = Object.keys(finalSettingsOfAllPublicSubjects).filter(key =>
         key.startsWith(InsightTypePrefix.langStats)
     ).length
 
-    const searchBasedInsights = Object.keys(finalSettingsOfAllOrgs).filter(key =>
+    const searchBasedExtensionInsightCount = Object.keys(finalSettingsOfAllPublicSubjects).filter(key =>
```
Comment (valerybugakov) [inline client/web/src/insights/core/analytics.ts:143]:
```suggestion
    const searchBasedExtensionInsightCount = Object.keys(finalSettingsOfAllPublicSubjects).filter(isSearchBasedInsightId).length
```

**Review by valerybugakov — APPROVED** [review]
LGTM! Added a couple of comments about reusing existing helper functions.

**Review by tjkandala — APPROVED** [review]

File: client/web/src/insights/core/analytics.ts (hunk 1 above)
Comment (Joelkw) [reply client/web/src/insights/core/analytics.ts:237]: Nice!

**Pull request merged**